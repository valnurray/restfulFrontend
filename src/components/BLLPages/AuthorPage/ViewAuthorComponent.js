import React, {Component} from "react";
import AuthorService from "../../../services/AuthorService"

class ViewAuthorComponent extends Component {
    constructor(props) {
        super(props)

        this.state = {
            id: this.props.match.params.id,
            author: {},
            article: {},
            articles: []
        }
    }

    componentDidMount() {
        AuthorService.getAuthorById(this.state.id).then(res => {
            this.setState({author: res.data});
        })
    }

    cancel() {
        this.props.history.push('/author');
    }

    render() {
        return (
            <div>
                <br></br>
                <div className="card col-md-6 offset-md-3">
                    <h3 className="text-center"> View Author Details</h3>
                    <div className="card-body">
                        <div className="row">
                            <label> Author ID: {this.state.author.id}</label>
                        </div>
                        <div className="row">
                            <label> Author Description: {this.state.author.description}</label>
                        </div>
                        <div className="row">
                            <label> Author Firs Name: {this.state.author.firstName}</label>
                        </div>
                        <div className="row">
                            <label> Author Last Name: {this.state.author.lastName}</label>
                        </div>
                    </div>

                </div>
                <button className="btn btn-danger" onClick={this.cancel.bind(this)} style={{marginLeft: "10px"}}>HOME
                </button>
            </div>
        )
    }
}




export default ViewAuthorComponent;