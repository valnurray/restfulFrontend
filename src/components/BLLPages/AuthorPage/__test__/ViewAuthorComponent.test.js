import React from "react";
import ReactDOM from 'react-dom'
import ViewAuthorComponent from '../ViewAuthorComponent';
import {cleanup, render} from "@testing-library/react";
import '@testing-library/jest-dom';
import renderer from "react-test-renderer";
import {match} from "react-router-dom";

describe("ViewAuthorComponent", () => {
    const match = {params : { id: 1}};

    afterEach(cleanup);


    it("render without crashing", () => {
        const div = document.createElement(`div`);
        ReactDOM.render(<ViewAuthorComponent match={match}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("render button cancel correctly", () => {
        const {getByTestId} = render(<ViewAuthorComponent match={match} />);
        expect(getByTestId('ViewAuthorComponent')).toHaveTextContent('View Author Details');
    });

    it('matches snapshot', () => {
        const tree = renderer.create(<ViewAuthorComponent match={match} />).toJSON();
        expect(tree).toMatchSnapshot();
    })

})