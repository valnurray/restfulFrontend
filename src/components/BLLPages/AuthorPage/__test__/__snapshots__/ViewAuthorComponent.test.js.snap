// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewAuthorComponent matches snapshot 1`] = `
<div
  data-testid="ViewAuthorComponent"
>
  <br />
  <div
    className="card col-md-6 offset-md-3"
  >
    <h3
      className="text-center"
    >
      View Author Details
    </h3>
    <div
      className="card-body"
    >
      <div
        className="row"
      >
        <label>
           Author ID: 
        </label>
      </div>
      <div
        className="row"
      >
        <label>
           Author Description: 
        </label>
      </div>
      <div
        className="row"
      >
        <label>
           Author Firs Name: 
        </label>
      </div>
      <div
        className="row"
      >
        <label>
           Author Last Name: 
        </label>
      </div>
    </div>
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    style={
      Object {
        "marginLeft": "10px",
      }
    }
  >
    HOME
  </button>
</div>
`;
